### PR TITLE
feature: improve scripts, add new version and upgrade repo scripts

### PR DESCRIPTION
**Pull request by clauBv23**
questions: 
Should we have a single script and receive as a parameter in the run function the operation?

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -2,54 +2,50 @@
...
-    {
+    function _createPluginRepo() internal returns (PluginRepo _sppRepo) {
         // create plugin repo
         _sppRepo = PluginRepoFactory(pluginRepoFactory).createPluginRepo(
             PluginSettings.PLUGIN_REPO_ENS_SUBDOMAIN_NAME,
             deployer
         );
 
-        _sppSetup = new SPPSetup();
+        if (_sppRepo == PluginRepo(address(0))) {
```
Comment (novaknole) [inline script/Deploy.s.sol:39]: Would this ever happen ?

File: script/UpgradeRepo.s.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        );
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        SPPSetup sppSetup;
+        if (isDeployerAllowed) {
+            // upgrade the repo
+            sppRepo.upgradeTo(address(latestBaseRepo));
+        } else {
+            sppSetup = new SPPSetup();
```
Comment (novaknole) [inline script/UpgradeRepo.s.sol:52]: What does this mean ? why are we deploying this ? maybe best thing would be to just console log that deployer doesn't have permission.

File: script/Deploy.s.sol (hunk 1 above)
Comment (clauBv23) [reply script/Deploy.s.sol:39]: probably never, should I remove the check?

Comment (novaknole) [reply script/Deploy.s.sol:39]: yes..

File: script/UpgradeRepo.s.sol (hunk 2 above)
Comment (clauBv23) [reply script/UpgradeRepo.s.sol]: yes, totally that's there by mistake. 
also added a log https://github.com/aragon/staged-proposal-processor-plugin/pull/27/commits/78d7825f38f8266cc04ee3c0bdf2bf475dbda019

**Pull request merged**